crates/iota-core/src/transaction_manager.rs

```diff
@@ -438,15 +438,34 @@ impl TransactionManager {
         let mut receiving_objects: HashSet<InputKey> = HashSet::new();
         let certs: Vec<_> = certs
             .into_iter()
-            .map(|(cert, fx_digest)| {
+            .filter_map(|(cert, fx_digest)| {
```

Comment:
Maybe add a TODO in line 423 that we need to add fatal there.

Reply:
Thanks for pointing this out! Fixed at [bec3e9b](https://github.com/iotaledger/iota/pull/4757/commits/bec3e9b139251a8fb5a5d6595c681c517ae9030d).